test/yz_pulseh.erl

```diff
@@ -0,0 +1,74 @@
+-module(yz_pulseh).
+-compile([export_all]).
+
+-ifdef(EQC).
+
+compile(Module) ->
+    compile(Module, []).
+
+compile(Module, UserOptions) ->
+    %% Trigger load if not present
+    _ = Module:module_info(),
+
+    %% Then work out where we loaded it from
+    case proplists:get_value(Module, code:all_loaded()) of
+        undefined ->
+            {error, not_loaded};
+        [$p, $u, $l, $s, $e, $d, $_ |  BeamName] ->
+            do_compile_beam(Module, BeamName, UserOptions);
+        BeamName ->
+            do_compile_beam(Module, BeamName, UserOptions)
+    end.
+
+%% Beam is a binary or a .beam file name
+do_compile_beam(Module,Beam,UserOptions) ->
+    %% Extract the abstract format and apply the pulse instrument
+    %% every executable line and, as a side effect, initiate
+    %% the database
+
+    case get_abstract_code(Module, Beam) of
+	no_abstract_code=E ->
+	    {error,E};
+	encrypted_abstract_code=E ->
+	    {error,E};
+	{_Vsn,Code} ->
+            Forms0 = epp:interpret_file_attribute(Code),
+	    Forms = pulse_instrument:parse_transform(Forms0, UserOptions),
+
+	    %% We need to recover the source from the compilation
+	    %% info otherwise the newly compiled module will have
+	    %% source pointing to the current directory
+	    SourceInfo = get_source_info(Module, Beam),
+
+	    %% Compile and load the result
+	    %% It's necessary to check the result of loading since it may
+	    %% fail, for example if Module resides in a sticky directory
+	    {ok, Module, Binary} = compile:forms(Forms, SourceInfo ++ UserOptions),
```

Comment:
@fadushin and I are running into a literal `error` here w/ compile:forms/2. We need to get solr_eqc.erl running before going ahead, as per https://github.com/basho/yokozuna/issues/602. I won't have time for this for a bit, but I wanted to jot this down. ^^ @jonmeredith